### PR TITLE
Add email docs

### PR DESCRIPTION
**Pull request by hishamco**

### CODE REVIEW

File: src/OrchardCore/OrchardCore.Email.Abstractions/ISmtpService.cs [hunk 1]
```diff
@@ -2,11 +2,16 @@
...
+    /// <summary>
+    /// Represents a contract for SMTP service.
+    /// </summary>
     public interface ISmtpService
     {
         /// <summary>
         /// Sends the specified message to an SMTP server for delivery.
         /// </summary>
+        /// <param name="message">The message to be sent.</param>
+        /// <returns>A <see cref="SmtpResult"/> that hold information about the sent message, for intsnace if it's sent successfully or it fails,</returns>
```
Comment (agriffard) [inline src/OrchardCore/OrchardCore.Email.Abstractions/ISmtpService.cs]: that hold`s`
for intsnac
if it has been sent successfully or if it has failed

File: src/OrchardCore/OrchardCore.Email.Abstractions/MailMessage.cs [hunk 2]
```diff
@@ -1,14 +1,48 @@
...
+        /// </summary>
         public string From {get; set;}
+
+        /// <summary>
+        /// Gets or sets the recipents.
+        /// </summary>
         public string To {get; set;}
+
+        /// <summary>
+        /// Gets or sets the caron copy emails.
```
Comment (agriffard) [inline src/OrchardCore/OrchardCore.Email.Abstractions/MailMessage.cs]: caron => carbon

File: src/OrchardCore/OrchardCore.Email.Abstractions/SmtpEncryptionMethod.cs [hunk 3]
```diff
@@ -4,6 +4,9 @@
 
 namespace OrchardCore.Email
 {
+    /// <summary>
+    /// Represents an enumeration for mail encrption methods.
```
Comment (agriffard) [inline src/OrchardCore/OrchardCore.Email.Abstractions/SmtpEncryptionMethod.cs]: encrption

File: src/OrchardCore/OrchardCore.Email.Abstractions/SmtpSettings.cs [hunk 4]
```diff
@@ -6,26 +6,68 @@
...
         public string DefaultSender { get; set; }
 
+        /// <summary>
+        /// Gete or sets the mail delivery method.
+        /// </summary>
         [Required]
         public SmtpDeliveryMethod DeliveryMethod { get; set; }
 
+        /// <summary>
+        /// Gets or sete the mailbox directory, this used for <see cref="SmtpDeliveryMethod.SpecifiedPickupDirectory"/> option.
```
Comment (agriffard) [inline src/OrchardCore/OrchardCore.Email.Abstractions/SmtpSettings.cs]: sete => set

**Comment by hishamco** [comment]
Can we merge this?

**Comment by hishamco** [comment]
@agriffard could you please merge this?

**Pull request merged**